jumpscale/sals/process/__init__.py

```diff
@@ -84,595 +141,833 @@ def execute(
-    else:
-        cmd = "ps ax | grep '{filterstr}'".format(filterstr=filterstr)
-    rc, out, err = execute(cmd)
-    # print out
-    found = []
-    for line in out.split("\n"):
-        if line.find("grep") != -1 or line.strip() == "":
-            continue
-        if line.strip() != "":
-            if line.find(filterstr) != -1:
-                line = line.strip()
-                if sortkey is not None:
-                    found.append(int([x for x in line.split(" ") if x][1]))
-                else:
-                    found.append(int(line.split(" ")[0]))
-    return found
+    ps_to_psutil_map = {
+        "%cpu": "cpu_percent",
+        "%mem": "memory_percent",
+        "cputime": "cpu_time",
+        "psr": "cpu_num",
+        "start_time": "create_time",
+        "egid": "egid",
+        "gid": "egid",
+        "euid": "euid",
+        "uid": "euid",
+        "euser": "username",
+        "pid": "pid",
+        "ppid": "ppid",
+    }
+    if sortkey is None:  # mimic default ps commnad sorting behavior
+        sortkey = "pid"
+    # return pids from process objects
+    return [p["pid"] for p in get_processes_info(sort=ps_to_psutil_map[sortkey], filterstr=filterstr, desc=desc)]
 
 
 def get_filtered_pids(filterstr, excludes=None):
-    """Get pids filtered by filterstr and execludes
-
-    Arguments:
-        filterstr {str} -- filter string.
+    """Get pids filtered by filterstr and excludes, matching against the full command line used to start the process.
 
-    Keyword Arguments:
-        excludes {list(str)} -- execlude list (default: {None})
+    Args:
+        filterstr (str): the String to filter based on.
+        excludes (list[str]): exclude list. Defaults to None.
 
     Returns:
-        [list(int)] -- pids
-    """
-    excludes = excludes or []
-    cmd = "ps ax | grep '%s'" % filterstr
-    rc, out, err = j.core.executors.run_local(cmd)
-    # print out
-    found = []
-
-    def checkexclude(c, excludes):
-        for item in excludes:
-            c = c.lower()
-            if c.find(item.lower()) != -1:
-                return True
-        return False
-
-    for line in out.split("\n"):
-        if line.find("grep") != -1 or line.strip() == "":
-            continue
-        if line.strip() != "":
-            if line.find(filterstr) != -1:
-                line = line.strip()
-                if not checkexclude(line, excludes):
-                    # print "found pidline:%s"%line
-                    found.append(int(line.split(" ")[0]))
-    return found
-
+        list of int: list of the processes IDs
+    """
+    pids = []
+    try:
+        for proc in psutil.process_iter(["name", "cmdline"]):
+            cmd_line = " ".join(proc.info["cmdline"])
+            if proc.info["cmdline"] and filterstr in cmd_line:
+                j.logger.debug(f"found filter string: {filterstr} in command line: {cmd_line}")
```

Comment:
IMHO, these inner debug logs like this can be removed after making sure everything works :laughing:

Reply:
Will keep logs for now until we have more test coverage.